\begin{abstract}
	\commentA{Multiple sequence alignment (MSA) is a basic step in many analyses in computational biology, including predicting the structure and function of proteins, orthology prediction and estimating phylogenies.}
	%In computational biology, the multiple sequence alignment (MSA) problem is the basic requirement for performing several tasks such as protein function prediction, phylogenetic tree estimation etc. 
	The objective of MSA is to infer the homology among the sequences of chosen species. \commentA{%There are some estimation and evaluation criteria for multiple sequence alignment. 
	Commonly, the MSAs are inferred by optimizing a single function or objective. The alignments estimated under one criterion may be different to the alignments generated by other criteria, inferring discordant homologies and thus leading to different evolutionary histories relating the sequences.}
	%There is no single optimization criterion (i.e. objective function) which directly leads towards this target.
	%Therefore, different types of criteria based on different assumptions are being used in the literature. 
	\commentM{In recent past, researchers have advocated for the multi-objective formulation of MSA, to address this issue, where multiple conflicting objective functions are being optimized simultaneously to generate a set of alignments. However, no theoretical or empirical justification with respect to a real-life application has been shown for a particular multi-objective formulation.}
	%and also they did not demonstrate the suitability of their approach in real-life application. 
	In this study, we investigate the impact of multi-objective formulation in the context of phylogenetic tree estimation. \commentM{Employing multi-objective metaheuristics, we demonstrate that trees estimated on the alignments generated by multi-objective formulation are substantially better than the trees estimated by the state-of-the-art MSA tools, including PASTA, MUSCLE, CLUSTAL, MAFFT etc. We also demonstrate that highly accurate alignments with respect to popular measures like sum-of-pair (SP) score and total-column (TC) score do not necessarily lead to highly accurate phylogenetic trees.}
	Thus in essence we ask the question whether a phylogeny-aware metric can guide us in choosing appropriate multi-objective formulations that can result in better phylogeny estimation. And we answer the question affirmatively through carefully designed extensive empirical study. As a by-product we also suggest a methodology for primary selection of a set of objective funstions for a multi-objective formulation based on the association with the resulting phylogenetic tree.
	%that carefully chosen multi-objective formulation can achieve statistically significant improvement, with respect to the accuracy of the phylogenetic trees estimated from these alignments, over the state-of-the-art MSA tools.
	
	%However, those good solutions are hidden inside a large collection of diverse solutions which is the final deliverable of any multi-objective formulation. Also, the MOEAs take a lot of time to produce results. We conclude that, without an effective scheme for extracting the good solutions from the solution pool and speeding up the runtime, the multi-objective formulation will not be applicable for solving real-life problems. 
\end{abstract}
